File: data-loader/src/main/java/domain/valueobject/Quantity.java
package domain.valueobject;

import java.util.Objects;
import java.util.Random;

public class Quantity {

    private int value;

    private static int MAX_VALUE = 10000;

    private final String QUANTITY_NEGATIVE_ERROR_MESSAGE =
            "Quantity must not me a negative number";

    public Quantity(int value) {
        this.validate(value);
        this.value = value;
    }

    private void validate(int value) {
        if (value < 0)
            throw new IllegalArgumentException(QUANTITY_NEGATIVE_ERROR_MESSAGE);
    }

    public int getValue() {
        return this.value;
    }

    public void setValue(int value){
        this.value = value;
    }

    public static Quantity random() {
        Random rand = new Random();
        return new Quantity(rand.nextInt(MAX_VALUE));
    }

    public boolean equals(Quantity other) {
        return this.value == other.getValue();
    }

}

File: front-end/src/domain/valueobject/quantity.ts
export class Quantity {

    private readonly _value: number;
    private static MAX_VALUE: number = 10000;

    constructor(value: number) {
        this.validate(value);
        this._value = value;
    }

    private validate(value: number): void {
        const QUANTITY_NEGATIVE_ERROR_MESSAGE: string = "Quantity must not be a negative number";
        if (value < 0) {
            throw new Error(QUANTITY_NEGATIVE_ERROR_MESSAGE);
        }
    }

    get value(): number {
        return this._value;
    }

    public static random(): Quantity {
        const rand = Math.floor(Math.random() * Quantity.MAX_VALUE);
        return new Quantity(rand);
    }
}
